### ISSUE TITLE
feature: Download private inventory files

### ISSUE THREAD

**Issue by connesy** [mkdocstrings/mkdocstrings#707]
### Is your feature request related to a problem? Please describe.
In my company, we have a project `Foo`, in which we build the documentation with `mkdocs` and `mkdocstrings`. The generated site and `objects.inv` is then pushed to a private GitHub repository to publish the documentation on a private site through GitHub Pages.

Now we are trying to build the documentation for another project `Bar`, which has references to objects in `Foo`, however, since the inventory file for `Foo` is placed in a private repo, it cannot be downloaded as part of `mkdocs build`.

### Describe the solution you'd like
It would be great to have some way to be able to either pass credentials and/or have them be read from e.g. environment variables or `.netrc` when downloading inventory files.

### Describe alternatives you've considered
Manually downloading the inventory file(s) and pointing to the local file path in `mkdocs.yml` when building the documentation. This becomes a real hassle when dealing with a team of developers having to manually download multiple inventory files and changing `mkdocs.yml` to point to their local file system.

We are really happy with `mkdocstrings`, this is just the missing piece for us :slightly_smiling_face:

**Comment by pawamoy** [comment]
Hi @connesy, thanks for the feature request. So you want to be able to do this:

```
plugins:
- mkdocstrings:
    handlers:
      python:
        import:
        - https://${CREDENTIALS}@domain.com/etc/objects.inv
```

...where `CREDENTIALS` is an env var with a value like `username:password`. Do I understand correctly?

**Comment by connesy** [comment]
Hi @pawamoy 

Yes, that would be a great solution! `CREDENTIALS` would have to be able to be either `username:password` or a single `api_token`, as is often used with e.g. GitHub.

**Comment by pawamoy** [comment]
Yes you'd be able to put whatever you want in the environment variable and build your URL with it :slightly_smiling_face: I'll go with a naive solution first and not do any escaping/URL-encoding.

**Comment by connesy** [comment]
@pawamoy I have previously tested adding a token directly to the URL, but `urllib` failed to parse the URL upon making the request. It took the `token@github.com` as being the host.

**Comment by pawamoy** [comment]
Looks like you're right:

```pycon
>>> urlparse("https://username:password@example.com/path")
ParseResult(scheme='https', netloc='username:password@example.com', path='/path', params='', query='', fragment='')
```

We can probably parse the URL, extract the username and password or the token, then rebuild the URL and pass the auth credentials accordingly.

**Comment by connesy** [comment]
Would you be interested in a PR if I give it a shot?

**Comment by pawamoy** [comment]
Sure, that'd be great!
